### PR TITLE
TypeScript-ify use pages

### LINKED ISSUE #959: Add types to `composables/use-pages.js`
## Description

Add type checking to `use-pages.js`. This module has zero untyped dependencies. Make sure to add it to `tsconfig.json`'s `include` list.

Note: You'll need to extend the `NuxtAppOptions` type in the `typings/nuxt__types/index.d.ts` definitions to add the `localePath` function to the `app` variable.

## Additional context

Part of an ongoing effort to add type checking to parts of the project that can be type checked.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #959 by @sarayourfriend

## Description

This PR converts `use-pages.ts` to TypeScript and adds the `localePath` property to `NuxtAppOptions` interface. Originally, this was also converting `useLoadMore` composable, but because there was an error in it preventing the app from being interactive, the changes were moved to a different PR.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: typings/nuxt__types/index.d.ts [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import '@nuxt/types'
 import type { Details as UADetails } from 'express-useragent'
+import type { NuxtI18nApi } from '@nuxtjs/i18n/types/vue'
 
 declare module '@nuxt/types' {
   export interface NuxtAppOptions {
     $ua: UADetails | null
+    localePath: NuxtI18nApi.localePath
```
Comment (sarayourfriend) [inline typings/nuxt__types/index.d.ts:8]: Huh, this should already be on this interface due to the `@nuxtjs/i18n` being included in the `tsconfig.json` `types` option :thinking:

Comment (obulat) [reply typings/nuxt__types/index.d.ts:8]: Ah, I probably added it before adding `@nuxtjs/i18n`, OR simply because the issue said to add it 😁

**Review by sarayourfriend — APPROVED** [review]
LGTM! What a breeze :grin:

**Review by zackkrida — APPROVED** [review]

**Comment by zackkrida** [comment]
A perfect PR! +0/-0 lines of code changed 😁

**Pull request merged**